test/test_lstm_cell.h

```diff
@@ -0,0 +1,159 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "gtest/gtest.h"
+#include "testhelper.h"
+#include "tiny_dnn/tiny_dnn.h"
+
+using namespace tiny_dnn::activation;
+
+namespace tiny_dnn {
+
+TEST(lstm, train) {
+  network<sequential> nn;
+  adagrad optimizer;
+
+  nn << recurrent_layer(lstm(3, 2, true), 1);
+  nn.weight_init(weight_init::xavier());
+
+  vec_t a(3), t(2), a2(3), t2(2);
+
+  // clang-format off
+    a[0] = 3.0f; a[1] = 0.0f; a[2] = -1.0f;
```

Comment:
`float_t` can be either `float` or `double`. It's controled at compilation time by `CNN_USE_DOUBLE`
check: https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/config.h#L88-L92